src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,246 @@
+_HEADER_KEYS_MAP["createdAt"] = "Creation Time"
+_HEADER_KEYS_MAP["findings"] = "Notes"
+
+name_option = click.option(
+    "--name", required=False, help="Name of the case.", default=None
+)
+assignee_option = click.option(
+    "--assignee", required=False, help="User UID of the assignee.", default=None
+)
+description_option = click.option(
+    "--description", required=False, help="Description of the case.", default=None
+)
+notes_option = click.option(
+    "--notes", required=False, help="Notes on the case.", default=None
+)
+subject_option = click.option(
+    "--subject", required=False, help="User UID of a subject of the case.", default=None
+)
+status_option = click.option(
+    "--status",
+    required=False,
+    help="Status of the case. `OPEN` or `CLOSED`",
+    default=None,
+)
+
+case_number_arg = click.argument("case-number", type=int)
+file_event_id_option = click.option(
+    "--event-id", required=True, help="File event id associated to the case."
+)
+
+
+@click.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def cases(state):
+    """For managing cases and events associated with cases."""
+    pass
+
+
+@cases.command()
+@click.argument("name", type=str)
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@sdk_options()
+def create(state, name, subject, assignee, description, notes):
+    """Create a new case."""
+    response = state.sdk.cases.create(
+        name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+    )
+    return response
+
+
+@cases.command()
+@case_number_arg
+@name_option
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@status_option
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+def _parse_cases(pages):
+    cases = [case for page in pages for case in page["cases"]]
+    if not cases:
+        click.echo("No cases found.")
+    return cases
+
+
+@cases.command("list")
+@click.option(
+    "--name",
+    required=False,
+    help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option(
+    "--subject", required=False, help="Filter by user UID of the subject of a case."
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option(
+    "--min-create-time",
+    required=False,
+    help="Fetch cases created after given date time.",
+)
+@click.option(
+    "--max-create-time",
+    required=False,
+    help="Fetch cases created before given date time.",
+)
+@click.option(
+    "--min-update-time",
```

Comment:
`min` and `max` times should be `--begin` (or `-b`)  and `--end` (`-e`) and use the familiar options, to be like the other options in the Code42 CLI.